includes/wikia/models/NavigationModel.class.php

```diff
@@ -50,61 +52,39 @@ class NavigationModel extends WikiaModel {
 
 	private $biggestCategories;
 	private $lastExtraIndex = 1000;
-	private $extraWordsMap = array(
+	private $extraWordsMap = [
 		'voted' => 'GetTopVotedArticles',
 		'popular' => 'GetMostPopularArticles',
 		'visited' => 'GetMostVisitedArticles',
 		'newlychanged' => 'GetNewlyChangedArticles',
 		'topusers' => 'GetTopFiveUsers'
-	);
+	];
 
-	private $forContent = false;
 	private $shouldTranslateContent = true;
 
-	private $useSharedMemcKey = false;
-
 	// list of errors encountered when parsing the wikitext
-	private $errors = array();
-
-	public function __construct( $useSharedMemcKey = false ) {
-		parent::__construct();
-
-		$this->useSharedMemcKey = $useSharedMemcKey;
-	}
+	private $errors = [];
 
 	/**
 	 * Return memcache key used for given message / variable
 	 *
-	 * City ID can be specified to return key for different wiki
-	 *
 	 * @param string $messageName message / variable name
-	 * @param int|bool $cityId city ID (false - default to current wiki)
+	 *
 	 * @return string memcache key
 	 */
-	private function getMemcKey( $messageName, $cityId = false ) {
-		if ( $this->useSharedMemcKey ) {
-			$wikiId = wfWikiID();
-		} else {
-			$wikiId = ( is_numeric($cityId)) ? $cityId : intval( $this->wg->CityId );
-
-			// fix for internal and staff (BugId:15149)
-			if ($wikiId == 0) {
-				$wikiId = $this->wg->DBname;
-			}
-		}
+	private function getMemcKey( $messageName ) {
+		$messageName = str_replace( ' ', '_', $messageName );
 
-		$messageName = str_replace(' ', '_', $messageName);
-
-		return wfSharedMemcKey( __CLASS__, $wikiId, $this->wg->Lang->getCode(), $messageName, self::version );
+		return wfMemcKey( __CLASS__, $this->wg->Lang->getCode(), $messageName, self::CACHE_VERSION );
 	}
 
-	public function clearMemc( $key = self::WIKIA_GLOBAL_VARIABLE, $city_id = false ){
-		$this->wg->Memc->delete(
-			$this->getMemcKey( $key, $city_id )
```

Comment:
We've (Jolek & Damian) decided, that on factory change cache can be cleaned manually and this hook is not needed anymore. And this function was not used in any other place.